tests: jest fails to start after a filtered pnpm install ("Module ts-jest in the transform option was not found")

### Expected Behavior

When trying to install dependencies and build for just one workspace (e.g. cdktn-cli), the tests still pass.

### Actual Behavior

When running `pnpm install --filter "cdktn-cli..." && pnpm --filter "cdktn-cli..." run build && pnpm --filter cdktn-cli exec jest`The tests fail with this error:
```
● Validation Error:

  Module ts-jest in the transform option was not found.
         <rootDir> is: /Users/djm/src/ext/cdk-terrain/packages/cdktn-cli

  Configuration Documentation:
  https://jestjs.io/docs/configuration

/Users/djm/src/ext/cdk-terrain/packages/cdktn-cli:
[ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL] Command failed with exit code 1: jest
```

### Steps to Reproduce

In a clean checkout:
1. pnpm install --filter "cdktn-cli..."
2. pnpm --filter "cdktn-cli..." run build
3. pnpm --filter cdktn-cli exec jest

### Versions

language: typescript
cdktn-cli: 0.24.0
node: v24.19.0
cdktn: null
cdktf: 0.21.0
constructs: null
jsii: null
terraform: 1.15.8
arch: arm64
os: darwin 25.6.0

### Migration Context

I used cdktf and did NOT see this issue there (possible regression)

### Upstream cdktf Issue

_No response_

### Providers

_No response_

### Gist

_No response_

### Possible Solutions

I raised https://github.com/open-constructs/cdk-terrain/pull/382 to fix this (just set the transform to an empty map in jest.preset.js)

### Workarounds

_No response_

### Anything Else?

I am running with --filter as I'm trying to update a distro package for cdktn-cli to the latest version, and don't want to bring in modules that aren't required

### References

_No response_

### Help Wanted

- [x] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/jest.preset.js
+++ b/jest.preset.js
@@ -10,6 +10,7 @@ process.env.CHECKPOINT_DISABLE = '1';
 
 module.exports = {
   ...nxPreset,
+  transform: {},
   testEnvironment: 'node',
   moduleFileExtensions: ['ts', 'tsx', 'js'],
   testMatch: ['**/*.test.ts', '**/*.test.tsx'],

